### ISSUE TITLE
A+ must not try to render binary files in plain text in the submitted file views

### ISSUE THREAD

**Issue by Mankro** [apluslms/a-plus#965]
Student's submitted files are usually opened in a modal dialog for a preview of the contents. In the teacher's inspect submission page, the preview is within the page without a modal dialog. The preview works for plain text files. A+ does not recognize all binary file types, thus A+ tries to render previews for some binary files too. That results in bad user experience because the preview shows only garbled text and non-printable characters.

A+ must not try to render binary files in plain text. Binary files may only be downloaded without a preview in the web page.

A+ already recognizes some binary file types, but their MIME types are hardcoded, which is not optimal:
https://github.com/apluslms/a-plus/blob/8811feab20862474ac69951ec04315a98e79c659/exercise/submission_models.py#L606

Microsoft Office files (.doc, .docx, .xls, .xlsx) are particularly missing. They are needed urgently.

If there were a better way to recognize binary files without listing them manually, that would be the best. There are libraries that try to recognize files types from the header of the file or try to check if the file contains only printable ASCII/Unicode characters/bytes.

Students see submitted files in submission feedback pages and modal dialogs. Course staff see submitted files in the inspect submission page in addition to the student participant overview and the submission feedback pages.

Related to #728.

Internal ticket:
https://rt.cs.aalto.fi/Ticket/Display.html?id=20621
